File: Api/Api/Domain/Model/Address.cs
using Api.Domain.Model.Base;

namespace Api.Domain.Model
{
    public class Address : BaseTable
    {
        public virtual int Id { get; protected set; }
        public virtual string Street { get; set; }
        public virtual int? Number { get; set; }
        public virtual string Neighborhood { get; set; }
        public virtual string City { get; set; }
        public virtual string State { get; set; }
        public virtual string Country { get; set; }
        public virtual string Extra { get; set; }
    }
}

File: App/lib/address/address.dart
class Address {
  String? id;
  String street;
  int number;
  String neighborhood;
  String city;
  String state;
  String country;
  String? extra;

  Address(
      {this.id,
      required this.street,
      required this.number,
      required this.neighborhood,
      required this.city,
      required this.state,
      required this.country,
      this.extra});

  static Address fromMap(Map<String, dynamic> map) {
    return Address(
        id: map['id'],
        street: map['street'],
        number: map['number'],
        neighborhood: map['neighborhood'],
        city: map['city'],
        state: map['state'],
        country: map['country'],
        extra: map['extra']);
  }
}
